Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -572,6 +591,47 @@ namespace Terrain
         sector.m_normalsBuffer->UpdateData(normals.data(), normals.size_bytes(), 0);
     }
 
+    void TerrainMeshManager::UpdateSectorLodBuffers(StackSectorData& sector, const AZStd::span<const HeightDataType> heights, const AZStd::span<const NormalDataType> normals)
+    {
+        // Store the height and normal information for the next lod level in each vertex for continuous LOD.
+        AZStd::vector<HeightDataType> lodHeights;
+        AZStd::vector<NormalDataType> lodNormals;
+        lodHeights.resize_no_construct(GridVerts1D * GridVerts1D);
+        lodNormals.resize_no_construct(GridVerts1D * GridVerts1D);
+
+        constexpr uint32_t LodGridVerts1D = (GridVerts1D >> 1) + 1;
+
+        for (uint32_t yPos = 0; yPos < GridVerts1D; ++yPos)
+        {
+            for (uint32_t xPos = 0; xPos < GridVerts1D; ++xPos)
+            {
+                uint32_t index = yPos * GridVerts1D + xPos;
+                uint32_t lodIndex1 = (yPos / 2) * LodGridVerts1D + (xPos / 2);
+                uint32_t lodIndex2 = lodIndex1;
+
+                if (xPos % 2 == 1)
+                {
+                    // x position is between two vertices in the row
+                    ++lodIndex1;
+                }
+                if (yPos % 2 == 1)
+                {
+                    // y position is between two vertices in the column
+                    lodIndex2 += LodGridVerts1D;
+                }
+                lodHeights[index] = (heights[lodIndex1] + heights[lodIndex2]) / 2;
+
+                // This is technically not correct because the normals are based off of neighbors from this lod, not the next lod.
+                lodNormals[index].first = (normals[lodIndex1].first + normals[lodIndex2].first) / 2;
+                lodNormals[index].second = (normals[lodIndex1].second + normals[lodIndex2].second) / 2;
```

Comment:
It would be nice to have a bit more explanation of why we're doing it if it's not correct. :P

Reply:
You found a note to myself that I meant to remove 😄 I should have made it uglier with a giant TODO or something.